src/main/java/org/jboss/msc/service/ServiceControllerImpl.java

```diff
@@ -1599,6 +1599,23 @@ private DependentsControllerTask() {
             dependents = getDependentsByDependencyName();
             children = ServiceControllerImpl.this.children.toScatteredArray(NO_DEPENDENTS);
         }
+
+        final boolean execute() {
+            for (Map.Entry<ServiceName, Dependent[]> dependentEntry : dependents.entrySet()) {
```

Comment:
If we're just iterating the map, then what we really want is a list of pairs, not a map.